src/tests/test_config.py

```diff
@@ -19,3 +45,263 @@ def test_dsn_filename_resolves_user_directory(sql_magic):
 
     # but it should resolve the path
     assert sql_magic.dsn_filename == expected
+
+
+def test_no_error_if_connection_file_doesnt_exist(tmp_empty, ip_no_magics):
+    ip_no_magics.run_cell("%config SqlMagic.dsn_filename = 'connections.ini'")
+
+    load_ipython_extension(ip_no_magics)
+
+    assert not Path("connections.ini").exists()
+
+
+def test_no_error_if_connection_file_doesnt_have_default_section(
+    tmp_empty, ip_no_magics
+):
+    Path("connections.ini").write_text(
+        """
+[duck]
+drivername = sqlite
+"""
+    )
+
+    ip_no_magics.run_cell("%config SqlMagic.dsn_filename = 'connections.ini'")
+
+    load_ipython_extension(ip_no_magics)
+
+    assert not ConnectionManager.connections
+
+
+def test_start_ini_default_connection_if_any(tmp_empty, ip_no_magics):
+    Path("connections.ini").write_text(
+        """
+[default]
+drivername = sqlite
+"""
+    )
+
+    ip_no_magics.run_cell("%config SqlMagic.dsn_filename = 'connections.ini'")
+
+    load_ipython_extension(ip_no_magics)
+
+    assert set(ConnectionManager.connections) == {"default"}
+    assert ConnectionManager.current.dialect == "sqlite"
+
+
+def test_start_ini_default_connection_using_pyproject_if_any(tmp_empty, ip_no_magics):
+    Path("pyproject.toml").write_text(
+        """
+[tool.jupysql.SqlMagic]
+dsn_filename = 'myconnections.ini'
+"""
+    )
+
+    Path("myconnections.ini").write_text(
+        """
+[default]
+drivername = duckdb
+"""
+    )
+
+    load_ipython_extension(ip_no_magics)
+
+    assert set(ConnectionManager.connections) == {"default"}
+    assert ConnectionManager.current.dialect == "duckdb"
+
+
+def test_magic_initialization_with_no_pyproject(tmp_empty, ip_no_magics):
+    load_ipython_extension(ip_no_magics)
+
+
+def test_magic_initialization_with_corrupted_pyproject(tmp_empty, ip_no_magics, capsys):
+    Path("pyproject.toml").write_text(
+        """
+[tool.jupysql.SqlMagic]
```

Comment:
the contents look correct of pyproject. why is this courrupted?

Reply:
the string isn't quoted